src/create-probot-app.ts

```diff
@@ -1,334 +1,24 @@
 #!/usr/bin/env node
-
-import fs from "fs-extra";
-import path from "path";
-import commander from "commander";
-import kebabCase from "lodash.kebabcase";
-import inquirer, { Answers, Question, QuestionCollection } from "inquirer";
-import validatePackageName from "validate-npm-package-name";
-import { guessEmail, guessAuthor, guessGitHubUsername } from "conjecture";
-import { generate } from "egad";
-
-import jsesc from "jsesc";
-import camelCase from "lodash.camelcase";
-import npm from "npm";
-import stringifyAuthor from "stringify-author";
-
+import { getAnswers, getProgram } from "./helpers/user-interaction";
 import { initGit } from "./helpers/init-git";
-import {
-  blue,
-  bold,
-  green,
-  printSuccess,
-  red,
-  writeHelp,
-  yellow,
-} from "./helpers/write-help";
-
-// TSC mangles output directory when using normal import methods for
-// package.json. See
-// https://github.com/Microsoft/TypeScript/issues/24715#issuecomment-542490675
-const pkg = require(require.resolve("../package.json"));
-
-/**
- * Partially sanitizes keys by escaping double-quotes.
- *
- * @param {Object} object The object to mutate.
- * @param {String[]} keys The keys on `object` to sanitize.
- */
-function sanitizeBy(
-  object: {
-    [key: string]: string;
-  },
-  keys: string[]
-): void {
-  keys.forEach((key) => {
-    if (key in object) {
-      object[key] = jsesc(object[key], {
-        minimal: true,
-        quotes: "double",
-      });
-    }
-  });
-}
-
-/**
- * Run `npm install` in `destination` folder, then run `npm run build`
- * if `toBuild` is true
- *
- * @param {String} destination Destination folder relative to current working dir
- * @param {Boolean}    toBuild Execute `npm run build` if true
- */
-function installAndBuild(destination: string, toBuild: boolean): Promise<void> {
-  class NpmError extends Error {
-    constructor(msg: string, command: string) {
-      const message = `
-
-Could not ${msg}.
-Try running ${bold("npm " + command)} yourself.
-
-`;
-      super(message);
-    }
-  }
+import { installAndBuild } from "./helpers/run-npm";
+import { makeScaffolding } from "./helpers/filesystem";
+import { printSuccess, red } from "./helpers/write-help";
 
-  return new Promise((resolve, reject) => {
-    const previousDir: string = process.cwd();
-    process.chdir(destination);
+function main(): void {
```

Comment:
GH doesn't support multi-line suggestions where lines have been removed... yet. This is the last one.


```suggestion
async function main(): Promise<void> {
```